ev3dev/control/GyroBalancer.py

```diff
@@ -1,398 +1,477 @@
+                # "the motor" angle, which is essentially how far the middle
+                # of the robot has traveled.
+                motorAngleRaw = 0
+
+                # The angle of the motor, converted to radians (2*pi radians
+                # equals 360 degrees).
+                motorAngle = 0
+
+                # The reference angle of the motor. The robot will attempt to
+                # drive forward or backward, such that its measured position
+                motorAngleReference = 0
+                # equals this reference (or close enough).
+
+                # The error: the deviation of the measured motor angle from the
+                # reference. The robot attempts to make this zero, by driving
+                # toward the reference.
+                motorAngleError = 0
+
+                # We add up all of the motor angle error in time. If this value
+                # gets out of hand, we can use it to drive the robot back to
+                # the reference position a bit quicker.
+                motorAngleErrorAccumulated = 0
+
+                # The motor speed, estimated by how far the motor has turned in
+                # a given amount of time.
+                motorAngularSpeed = 0
+
+                # The reference speed during manouvers: how fast we would like
+                # to drive, measured in radians per second.
+                motorAngularSpeedReference = 0
+
+                # The error: the deviation of the motor speed from the
+                # reference speed.
+                motorAngularSpeedError = 0
+
+                # The 'voltage' signal we send to the motor.
+                # We calculate a new value each time, just right to keep the
+                # robot upright.
+                motorDutyCycle = 0
+
+                # The raw value from the gyro sensor in rate mode.
+                gyroRateRaw = 0
+
+                # The angular rate of the robot (how fast it is falling forward
+                # or backward), measured in radians per second.
+                gyroRate = 0
+
+                # The gyro doesn't measure the angle of the robot, but we can
+                # estimate this angle by keeping track of the gyroRate value in
+                # time.
+                gyroEstimatedAngle = 0
+
+                # Over time, the gyro rate value can drift. This causes the
+                # sensor to think it is moving even when it is perfectly still.
+                # We keep track of this offset.
+                gyroOffset = 0
+
+                eprint("Hold robot upright. Press Touch Sensor to start.")
+
+                # Wait for Touch Sensor or any Button Press
+                while not self.touchSensor.is_pressed:
+                    time.sleep(0.01)
+
+                # Wait for release and proceed to calibration and balancing
+                while self.touchSensor.is_pressed:
+                    time.sleep(0.01)
```

Comment:
You can use `self.touchSensor.wait_for_bump()` here (in place of lines 258->264)